internal/hardware/validator.go

```diff
@@ -130,6 +131,42 @@ func (v *validator) GetHostRequirements(role models.HostRole) models.HostRequire
 	}
 }
 
-func IsVirtSupported(inventory *models.Inventory) bool {
-	return funk.Contains(inventory.CPU.Flags, intelVirtCpuFlag) || funk.Contains(inventory.CPU.Flags, amdVirtCpuFlag)
+func (v *validator) GetClusterHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirements, error) {
+	operatorsRequirements, err := v.operatorsAPI.GetRequirementsBreakdownForRole(ctx, cluster, host.Role)
```

Comment:
So now because `operatorsAPI.GetRequirementsBreakdownForRole` doesn't work with the AutoAssign role, there won't be any minimum requirements.  Is that the expected behavior?

Reply:
The reason for ignoring AutoAssign role was CNV validation, but you're right - when CNV operator is enabled and requirements are retrieved for a host with AutoAssign, there will be error originating from CNV operator plugin and I think it might be generated even before cnv operator validation is executed, which rejects  `AutoAssign` role. 

I think it makes more sense to support AutoAssign in that one particular part of CNV operator code and treat is the same as Worker (as I saw being done in other places in the code).

Reply:
cc @pkliczewski

Reply:
As far as I understand not only CNV would [reject](https://github.com/openshift/assisted-service/blob/master/internal/operators/ocs/validations.go#L87) `AutoAssign` role but OCS as well.

Reply:
That is at the level of validation, not requirements providing; it won't change with adding support for `AutoAssign` in `api.Operator.GetHostRequirementsForRole`.